public/pages/DetectorResults/containers/AnomalyResultsTable.tsx

```diff
@@ -62,6 +67,99 @@ export function AnomalyResultsTable(props: AnomalyResultsTableProps) {
       ? props.anomalies.filter((anomaly) => anomaly.anomalyGrade > 0)
       : [];
 
+  const handleOpenDiscover = async (startTime: number, endTime: number, item: any) => {
+    try {
+      // calculate time range with 10-minute buffer on each side per customer request
+      const TEN_MINUTES_IN_MS = 10 * 60 * 1000;
+      const startISO = new Date(startTime - TEN_MINUTES_IN_MS).toISOString();
+      const endISO = new Date(endTime + TEN_MINUTES_IN_MS).toISOString();
+      
+      const basePath = `${window.location.origin}${window.location.pathname.split('/app/')[0]}`;
```

Comment:
instead you could use the core service that is accessible by every plugin `getUrlForApp` 
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/core/public/application/types.ts#L826

so like
```typescript
window.open(getUrlForApp(def.appId), '_blank', 'noopener, noreferrer')
```

```

Reply:
Thanks for the suggestion, I didn't know we have such method :) 
I would leave it as the current implementation because these two methods work exactly the same, but the one that uses the core service adds extra burden when writing unit tests, as it needs to be mocked.

Reply:
if you plan on mock the services to be super basic, you should be able to use the helper methods which should instantiate all the mocks for those services

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/core/public/mocks.ts#L200

for example,
```typescript
import { coreMock } from '../../../../../core/public/mocks';

const startMock = coreMock.createStart();
```